account_asset_analytic/models/account_invoice_line.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class AccountInvoceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    @api.model
+    def asset_create(self, lines):
+        res = super(AccountInvoceLine, self).asset_create(lines)
+        for line in lines:
+            if line.account_analytic_id:
+                asset = self.env['account.asset.asset'].search([
+                    ('code', '=', line.invoice_id.number),
+                    ('company_id', '=', line.company_id.id),
+                ])
```

Comment:
put `limit=1` (not recommended, because you can confirm, cancel and confirm again, and have 2 assets) or call it `assets` and make a write instead an active record assignation.